noir/noir-repo/acvm-repo/acir/src/circuit/mod.rs

```diff
@@ -93,10 +87,55 @@ pub enum AssertionPayload {
     Dynamic(/* error_selector */ u64, Vec<ExpressionOrMemory>),
 }
 
+#[derive(Debug, Copy, PartialEq, Eq, Hash, Clone, PartialOrd, Ord)]
+pub struct ErrorSelector(u64);
+
+impl ErrorSelector {
+    pub fn new(integer: u64) -> Self {
+        ErrorSelector(integer)
+    }
+
+    pub fn as_u64(&self) -> u64 {
+        self.0
+    }
+}
+
+impl Serialize for ErrorSelector {
```

Comment:
It's serialized as string since the Serde<>JsValue bridge that we use serializes u64s as JS numbers, which don't have enough precision to store u64s.